### ISSUE TITLE
Tracker Data Entry Improvements

### ISSUE THREAD

**Issue by burnsauce** [monome/teletype#63]
The data entry on the tracker screen should be buffered, i.e.: users should have to press 'ENTER' to confirm data, and could hit 'ESC' to cancel entry.

It would permit users to enter large values into patterns instantaneously.  Currently, intermediary typed values are caught by scripts reading the patterns at high rates -- any rate higher than the data entry speed.

Additionally, this would eliminate the problem of needing to delete the existing data in the field before entering data.

**Comment by burnsauce** [comment]
As ESC is reserved, the implementation put forth uses all non-data-entry, yet valid keystrokes to abort data entry.

My only gripe is that I selected _append-on-entry_ behaviour for nonzero pattern values instead of _clear-on-entry_, as I would have preferred.  I selected the append behaviour because it followed the existing data entry model and it might be muscle memory for some users.  It's an easy fix if we agree that clear-on-entry is the better model.
